File: tests/fixtures/treesitter/sample.go
package sample

import (
	"fmt"
	"net/http"
)

// DefaultTimeout is the default request timeout.
const DefaultTimeout = 30

// MaxRetries controls the retry limit.
var MaxRetries = 3

// unexported constant (should not appear in exports)
const internalLimit = 100

// Config holds application configuration.
type Config struct {
	Host string
	Port int
}

// Handler defines the request handler interface.
type Handler interface {
	ServeHTTP(w http.ResponseWriter, r *http.Request)
}

// StringAlias is a named type.
type StringAlias string

// NewConfig creates a new Config with defaults.
func NewConfig(host string, port int) *Config {
	return &Config{Host: host, Port: port}
}

// unexported function (should not appear)
func helperFunc() {}

// Start launches the server.
func (c *Config) Start() error {
	fmt.Printf("Starting on %s:%d\n", c.Host, c.Port)
	return nil
}

// unexported method (should not appear)
func (c *Config) validate() bool {
	return c.Host != ""
}

File: tests/fixtures/treesitter/sample.py
"""Sample module for testing Python tree-sitter extraction."""

import os
import sys
from pathlib import Path
from typing import Optional
from collections.abc import Callable


MAX_RETRIES = 3
DEFAULT_TIMEOUT = 30
_private_var = "hidden"


class Config:
    """Holds application configuration."""

    def __init__(self, host: str, port: int = 8080):
        self.host = host
        self.port = port

    def start(self) -> None:
        print(f"Starting on {self.host}:{self.port}")

    def _validate(self) -> bool:
        return self.host != ""


class _InternalHelper:
    """Should not appear in exports."""

    pass


class Handler(Config):
    """Extends Config with request handling."""

    def handle(self, request: bytes) -> bytes:
        return b"OK"


def create_config(host: str, port: int = 8080) -> Config:
    """Create a new Config with defaults."""
    return Config(host, port)


def _helper():
    pass


async def async_fetch(url: str, timeout: int = 30) -> bytes:
    """Fetch data asynchronously."""
    pass


from dataclasses import dataclass


@dataclass
class Settings:
    """Application settings as a dataclass."""

    debug: bool = False
    verbose: bool = False


CONSTANT_TUPLE = (1, 2, 3)
